[feat/#10] Metrics Server

bconfiden2 opened the pull request:
## 작업 내용

- [x] 메트릭 서버가 없을 경우를 대비한 metrics-server.yaml 추가
- [x] 파드 내에서 Metrics API 요청을 위한 rbac.yaml (클러스터 롤) 추가

## 참고 사항

bconfiden2 committed:
[feat/#10] Add: metrics server yaml file

yaml file for creating metrics server pod in kube-system namespace. Crea
tes ServiceAccount, ClusterRole, ClusterRole, ClusterRoleBinding, Servic
e, Deployments, etc.

bconfiden2 committed:
[feat/#10] Add: rbac.yaml

Creates Role-Based Access Control for accessing API server. Allows defau
lt ServiceAccount in default namespace

hellouz818 commented:
이 파일들이 어떤식으로 작동하는지에 대해 처음보는 팀원은 잘 모를 수 있기때문에 메트릭 서버 사용법에 대한 문서의 필요성을 느낍니다.
그래서 관련 내용에 대한 파일을 만들어서 이 PR에 이어서 커밋할 예정!

bconfiden2 commented:
[여기](https://kubernetes.io/ko/docs/tasks/run-application/access-api-from-pod/)를 참고했습니다~

Commit:
[feat/#10] Doc: metrics server manual

how to run metrics server

hellouz818 reviewed (commented):
코드 실행되는지 확인했고, 해당 실행방법 문서로 정리해서 올렸습니다!

(위의 문서인데, git config를 로컬에서 설정하지 않고 올려서 작성자가 없이 뜨네요,,ㅎ 다음 커밋부터는 제대로 올라갈거에요!)

bconfiden2 committed:
[feat/#10] Doc: manual revision

To not be confused between APISERVER / KUBERNETS_SERVICE_HOST

asd3638 commented:
좋아요

NayeonKeum commented on Collect/rbac.yaml:
```diff
@@ -0,0 +1,27 @@
+  name: view-metrics-aihpa
+subjects:
+- kind: ServiceAccount
+  name: default        # 테스트 해본 것은 아니지만, 해당 ServiceAccount 이름으로만 접근 가능할 것으로 보임
```
주석 지워도 될 것 같습니당

hellouz818 replied:
좋아요

NayeonKeum commented on Collect/rbac.yaml:
```diff
@@ -0,0 +1,27 @@
+subjects:
+- kind: ServiceAccount
+  name: default        # 테스트 해본 것은 아니지만, 해당 ServiceAccount 이름으로만 접근 가능할 것으로 보임
+  namespace: default   # 해당 네임스페이스에서 생성된 파드 안에서만 접근 가능 
```
얘도요!

hellouz818 replied:
아하! 제가 지워서 다시 커밋하겠습니다-!

asd3638 reviewed (commented):
굿

hellouz818 committed:
[feat/#10] Style: Delete comment

Delete korean description comment

hellouz818 committed:
Merge branch 'feat/#10' of https://github.com/Scooter-2022/AI-HPA into feat/#10

bconfiden2 commented:
ㅋㅋㅋㅋ 우리 앞으로 커밋하기 전에 반드시 pull 로 땡겨오는 습관을 들이자.. 원격이랑 로컬 충돌나서 머지하는거 지양

깜박하고 충돌난다? 그러면 싹 지우고 다시 받아와서 새로 커밋하기!

hellouz818 commented:
미안합니다,,^^,, 저도 패치 안하고 커밋한뒤에서 푸쉬하려니 컨플릭트나서 깜짝놀라버렸네요,,하핫,,,
담부턴 주의하고 푸쉬하겠습니다 또하나 배워갑니다,,ㅎㅎ

다음보는분이 PR 닫아주세요!

bconfiden2 committed:
[feat/#10] Fix: rename rbac to rbac-metrics

rename rbac.yaml file to rbac-metrics.yaml not to be confused by rbac-sc
ale.yaml file in Update module.

bconfiden2 committed:
[feat/#10] Fix: README.md

updated README.md as rbac.yaml file name has been changed.
